tools/clang/test/HLSL/ShaderOpArith.xml

```diff
@@ -3655,6 +3691,17 @@ void MSMain(uint GID : SV_GroupIndex,
 
     ROOT_SIG
     float4 PSMain(PSInput input) : SV_TARGET {
+    
+        int x = (int)input.position.x;
+        int y = (int)input.position.y;
+        int ix = x+y;
+        
+        ByteAddressBuffer g_rawBuf = ResourceDescriptorHeap[INDEXER(CB.idx0) + OFFSET(ix)];
+        StructuredBuffer<float> g_structBuf = ResourceDescriptorHeap[INDEXER(CB.idx2) + OFFSET(ix)];
+        Texture2D<float> g_tex = ResourceDescriptorHeap[INDEXER(CB.idx4) + OFFSET(ix)];
+        RWByteAddressBuffer g_rwRawBuf = ResourceDescriptorHeap[INDEXER(CB.idx6) + OFFSET(ix)];
+        RWStructuredBuffer<float> g_rwStructBuf = ResourceDescriptorHeap[INDEXER(CB.idx8) + OFFSET(ix)];
+        RWTexture1D<float> g_rwTex = ResourceDescriptorHeap[INDEXER(CB.idx10) + OFFSET(ix)];
         
         g_resultPS[0] = g_rawBuf.Load<float>(0);
```

Comment:
If the resource is non-uniform, the value loaded here will be non-uniform, so you'll need to change the location written based on your `OFFSET(ix)` value so you can capture the results independently.  You probably want to add `(OFFSET(ix)*8)` since you have 8 result values per index read.  Then you'll need to expand and modify the corresponding verification code in the test.  Since you'll be adding more expected result values, at this point, might I suggest an expected result array that you loop through along with the output values, instead of a bunch of independent `VERIFY_ARE_EQUAL` statements?